modules/internal/String.chpl

```diff
@@ -464,6 +472,7 @@ module String {
   */
   inline proc createStringWithBorrowedBuffer(s: string) {
     var ret: string;
+    try! validateEncoding(s.buff, s.len);
```

Comment:
Do you think that should happen for this specific overload of `createStringWithBorrowedBuffer` or other overloads as well? How about other `createString*`?

It may be right have all of these throw, but I wanted to halt as a first step. My main concern being `createStringWithBorrowedBuffer` being used in many different places such as `Path` and `FileSystem` module. For which we have a different plan to support non UTF8 strings. And I hope to do that soon. But I didn't want to add bunch of `try/catch`es or `try!`s in those places only to replace them with something else.

That being said, this is not a strong opinion and I can do that.

Reply:
I'd be OK having the `try!` here transitionally but if that is the plan, you should create an issue documenting the future work, or at least mention it in the PR message.

In this specific case I'd advocate for making the `createString*` methods `throw` and then updating the call sites of them with `try!` and working over time to migrate those to better handling.